lib/context_adapter_helper.js

```diff
@@ -205,6 +217,71 @@ function isAttributeIncluded(attributes, attributeName) {
 }
 
 /**
+ * Validates the attribute values received
+ * @param {object} context The context for logging
+ * @param {string} entityType The entity type (Button or Service)
+ * @param {Array} attributes The attributes array
+ * @return {*} True if the attribute value is valid, false otherwise
+ */
+function isValidAttributeValues(context, entityType, attributes) {
+  var interactionTypeAttrName,
+    interactionTypes;
+  if (entityType === caConfig.BUTTON_ENTITY.TYPE) {
+    interactionTypeAttrName = caConfig.BUTTON_ENTITY.CA_INTERACTION_TYPE_ATTR_NAME;
+    interactionTypes = caConfig.BUTTON_ENTITY.INTERACTION_TYPES;
+  } else if (entityType === caConfig.SERVICE_ENTITY.TYPE) {
+    interactionTypeAttrName = caConfig.SERVICE_ENTITY.INTERACTION_TYPE_ATTR_NAME;
+    interactionTypes = caConfig.SERVICE_ENTITY.INTERACTION_TYPES;
+  } else {
+    caLogger.debug(
+      context,
+      'Invalid entity type when validating attribute values: ' + entityType
+    );
+    return false;
+  }
+
+  for (var i = 0; i < attributes.length; i++) {
+    if (attributes[i].name === interactionTypeAttrName) {
```

Comment:
Is this ok? two ifs and no else branch... so... could they be merged into just one if sentence?